code-challenges.js

```diff
@@ -3,43 +3,119 @@
 
+// create a function that takes a single number as an argument
+const divideByThree = (number) => {
+    // use %3 to see if a number can be divided by 3 with no remainder === 0
+    // check to see if user input a number
+    if (typeof number === "number") {
+        if (number % 3 === 0) {
+            // if num %3 === 0, it's divisible by three
+            return `${number} is dibisible by three`
+        }
+        else if (number % 3 !== 0) {
+            // if num %3 !== 0, its not divisble by three
+            return `${number} is not divisible by three`
+        }
+        else {
+            return "There has been some input error"
+        }
+    }
+    // if typeof number is not number, return error message
+    else {
+        return "There has been an input error. Did you enter a number?"
+    }
+}
+
+
 var num1 = 15
 // Expected output: "15 is divisible by three"
+console.log(divideByThree(num1));
 
 var num2 = 0
 // Expected output: "0 is divisible by three"
+console.log(divideByThree(num2));
 
 var num3 = -7
 // Expected output: "-7 is not divisble by three"
+console.log(divideByThree(num3));
 
+var string = "words"
+// my own test to see if the error message for type occurs
+console.log(divideByThree(string));
 
-
+var anotherValue = true
+// my own test to see if the error message for type occurs
+console.log(divideByThree(anotherValue));
 
 
 // --------------------2) Create a function that takes in the randomNouns variable and returns an array with all the words capitalized.
 
+//create a function that takes in an array of nouns
+const capitalizer = (array) => {
+    let capitalizedWords = (array.map(currentWord => {
+        // need to hold each word alone as current word as its own array
+        let arrayOfLetters = currentWord.split("")
+        // make a var to hold capitlized index 0 of array of letters
+        let capitalizedFirstLetter = arrayOfLetters[0].toUpperCase()
+        // keep a var of the slice after capitalizedFirstLetter and join to make a string again
+        let restOfWord = arrayOfLetters.slice(1).join("")
+        // create a variable that holds complete word
+        let completedWord = capitalizedFirstLetter + restOfWord
+        return completedWord
+    }))
+    //return the new array with all the completed words
+    return capitalizedWords
+}
+
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
-
-
+console.log(capitalizer(randomNouns));
 
 
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
+// create a function that takes in an array
+const numbersLeastToGreatest = (array) => {
+    //filter for typeof number, and push that into a new array
+    let numbersOnly = (array.filter(value => typeof value === "number"))
```

Comment:
So i'm curious why the value of numbersOnly is wrapped in parenthesis?

In JavaScript, the functions wrapped with parenthesis are called “Immediately Invoked Function Expressions" or "Self Executing Functions. The purpose of wrapping is to namespace and control the visibility of member functions. It wraps code inside a function scope and decrease clashing with other libraries.

Since this variable is not a stand alone function, nor are we working with outside libraries, I don't think it is necessary or really best practice to be doing this.

Reply:
Thank you so much for your feedback, Andee! I didn't do this for any particular reason. It's honestly just a typo, or a misunderstanding of what I need to enclose. Thank you for making clear to me why I should not do that!